Issue 883 - added docs not being included in model state update

boikle opened the pull request:
Fixed issue with variable values from previous loop iterations persisting and causing new documents from being added when the model state is updated.

This bug was preventing models with the type couchDbView from correctly updating when new docs are added to a model (and consequently also the map). 

Fix #883

sruest commented on nunaliit2-js/src/main/js/nunaliit2/n2.couchDbPerspective.js:
```diff
@@ -871,25 +872,29 @@ var ModelCouchDbView = $n2.Class({
 			} else {
-				// This is a removed document
-				removed.push(oldDoc);
+				if( oldDoc ){
```
Could collapse to 'else if'

sruest commented on nunaliit2-js/src/main/js/nunaliit2/n2.couchDbPerspective.js:
```diff
@@ -871,25 +872,29 @@ var ModelCouchDbView = $n2.Class({
 		// Detect added documents
-		for(var docId in docInfoMap){
-			var oldDoc;
+		for(docId in docInfoMap){
```
Is it possible to collapse these for loops into one?  Instead of iterating over the docInfoMap twice?

boikle replied:
I thought so as well, but then I noticed that the objects being iterated over are different. That's why they were separated in the code originally.

boikle commented:
Thank you for pointing out the issues above. 

I've refactored the code, and cleaned up all of the condition statements. Also looked into the two for loops, but they need to separate since they are iterating over different objects. 

I also improved object property checks to use the more defensive Object.hasOwnProperty.call() instead of objectName[propName].

boikle commented:
This pull request is related to issue #871
